.github/workflows/scripts/e2e-verify.sh

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+
+# To test:
+# export GITHUB_SHA=6f3b6435f5a17a25ad6cf2704d0c192bcef8193f
+# export GITHUB_RUN_ID=2272442563
+# export GITHUB_ACTOR=laurentsimon
+# export GITHUB_RUN_ATTEMPT=1
+# export GITHUB_REF=refs/heads/branch-name or refs/tags/tag-name
+# export GITHUB_REF_TYPE=branch or tag
+# export GITHUB_REPOSITORY=slsa-framework/example-package
+# export GITHUB_REF_NAME=v1.2.3
+# export GITHUB_WORKFLOW=go schedule main SLSA3 config-noldflags
+# export THIS_FILE=e2e.go.workflow_dispatch.main.config-noldflags.slsa3.yml
+# export BINARY=binary-linux-amd64
+# export PROVENANCE=example.intoto.jsonl
+
+source "./.github/workflows/scripts/e2e-utils.sh"
+
+# Provenance content verification.
+ATTESTATION=$(cat "$PROVENANCE" | base64 -d)
+LDFLAGS=$(echo "$THIS_FILE" | cut -d '.' -f3 | grep -v noldflags)
+
+e2e_verify_predicate_subject_name "$ATTESTATION" "binary-linux-amd64"
+e2e_verify_predicate_builder_id "$ATTESTATION" "https://github.com/Attestations/GitHubHostedActions@v1"
+e2e_verify_predicate_builderType "$ATTESTATION" "https://github.com/slsa-framework/slsa-github-generator-go@v1"
+
+#TODO(#108): don't set GITHUB_WORKFLOW.
+GITHUB_WORKFLOW="%!s(<nil>)"
+e2e_verify_predicate_invocation_configSource "$ATTESTATION" "{\"uri\":\"git+https://github.com/$GITHUB_REPOSITORY@$GITHUB_REF\",\"digest\":{\"sha1\":\"$GITHUB_SHA\"},\"entryPoint\":\"$GITHUB_WORKFLOW\"}"
+
+e2e_verify_predicate_invocation_environment "$ATTESTATION" "github_actor" "$GITHUB_ACTOR"
+e2e_verify_predicate_invocation_environment "$ATTESTATION" "github_sha1" "$GITHUB_SHA"
+e2e_verify_predicate_invocation_environment "$ATTESTATION" "os" "ubuntu20"
+e2e_verify_predicate_invocation_environment "$ATTESTATION" "arch" "X64"
+e2e_verify_predicate_invocation_environment "$ATTESTATION" "github_event_name" "$GITHUB_EVENT_NAME"
+e2e_verify_predicate_invocation_environment "$ATTESTATION" "github_ref" "$GITHUB_REF"
+e2e_verify_predicate_invocation_environment "$ATTESTATION" "github_ref_type" "$GITHUB_REF_TYPE"
+
+if [[ -z "$LDFLAGS" ]]; then
+    e2e_verify_predicate_buildConfig_command "$ATTESTATION" "[\"build\",\"-mod=vendor\",\"-trimpath\",\"-tags=netgo\",\"-o\",\"binary-linux-amd64\",\"./pr-e2e/go/main.go\"]"
+else
+    e2e_verify_predicate_buildConfig_command "$ATTESTATION" "[\"build\",\"-mod=vendor\",\"-trimpath\",\"-tags=netgo\",\"-ldflags=-X main.gitVersion=v1.2.3 -X main.gitCommit=abcdef -X main.gitBranch=$BRANCH\",\"-o\",\"binary-linux-amd64\",\"./pr-e2e/go/main.go\"]"
+    chmod a+x ./"$BINARY"
```

Comment:
You already expect that the binary name is `binary-linux-amd64` elsewhere (e.g. the subject, buildConfig.command etc.) so why not just use that static value here here?

Reply:
Can do that, thanks.

Reply:
on second thought, how about replacing all occurrences of `binary-linux-amd64` with `$BINARY`? This way the verification script is more generic?

Today we have a single workflow, but we may have more in the future, including for other ecosystems.
Note: there are still a bunch of hardcoded values (env variables and command) but we could make these dynamic later.

Wdut?

Reply:
Yeah, that's probably better.